app/models/user.rb

```diff
@@ -1,14 +1,12 @@
 class User < ApplicationRecord
   has_many :posts, foreign_key: 'author_id'
-  has_many :comments
-  has_many :likes
-
-  validates_presence_of :name
-  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
 
   after_create :recent_posts
 
-  def recent_posts
-    posts.order(created_at: :desc).limit(3)
+  # Method to get the 3 most recent posts for a user
+  def recent_posts(limit = 3)
+    posts.order(created_at: :desc).limit(limit)
   end
```

Comment:
Kindly implement this required changes from the previous reviewer. It is yet to be implemented. Kindly check the [differences between after_create and after_save](https://www.altcademy.com/questions/after-create-vs-after-save-in-rails#:~:text=after_create%20%2D%20works%20once%2C%20only%20after,create%20and%20ActiveRecord%23update%20method.) as explained to you by the previous reviewer.